horace_core/sqw/@sqw/noisify.m

```diff
@@ -0,0 +1,39 @@
+function wout = noisify(w,varargin)
+%=========================================================================
+% Adds random noise to the signal(s) of an sqw object or array of sqw
+% objects, together with an additional fixed error bar. Sqw objects are
+% noisified through their paged PixelData sub-objects. Alternatively 
+% noisifies a dnd object or array of such objects directly.
+%
+% Syntax:
+%   >> wout = noisify (w)
+%           Add noise with Gaussian distribution, with standard deviation
+%           = 0.1*(maximum pixel signal value)
+%
+%   >> wout = noisify (w,factor)
+%           Add noise with Gaussian distribution, with standard deviation
+%           = factor*(maximum pixel signal value)
+%
+%   >> wout = noisify (w,'poisson')
+%           Add noise with Poisson distribution, where the mean value at
+%           a point is the value of pixel signal.
+%
+% Additional developer options are possible in varargin to test this
+% functionality. See the PixelData noisify paging method (called below)
+% and the Herbert noisify function which implements the noise addition.
+%=========================================================================
+
+wout=w;
+for i=1:numel(w)
+    if isa(w(i),'sqw')   % determine if sqw or dnd type
```

Comment:
this is an SQW class method, do you need the check for DnD ness? The DnD version belongs on DnDBase

Reply:
Reviewing my previous comment ... should this be a check for `has_pixels(w(i))`? No DnD objects (should) be passed in here, and if a pixel-less SQW object is called the `noisify`/`recompute_bin_data` action will fail.